cmd/agent/daemon/enrichment/enrichers.go

```diff
@@ -112,7 +129,12 @@ func (enricher *fileHashEnricher) calcFileHashForPID(cont *containers.Container,
 	path := filepath.Join(pidString, "root", execPath)
 	info, err := enricher.procFS.Stat(path)
 	if err != nil {
-		// If the wanted file does not exist inside the mount namespace, there is also nothing we can do.
+		_, pathErr := enricher.procFS.Stat(pidString)
+		if pathErr == nil {
+			// If the /proc/<pid> folder exists, but the wanted file does not, we can assume it was deleted
+			return nil, ErrFileDeleted
```

Comment:
This does not fully work here, as the same can happen on line `149`, as well as `154` hmm